src/v0/destinations/intercom_v2/transform.js

```diff
@@ -0,0 +1,193 @@
+};
+
+const constructIdentifyResponse = async (event) => {
+  const { destination, metadata } = event;
+  const apiVersion = getApiVersion(destination);
+
+  const payload = transformIdentifyPayload(event);
+
+  let method = 'POST';
+  let endpoint = `${getBaseEndpoint(destination)}/contacts`;
+  const headers = getHeaders(metadata, apiVersion);
+
+  // when contact is found in intercom
+  const contactId = await searchContact(event);
+  if (contactId) {
+    method = 'PUT';
+    endpoint += `/${contactId}`;
+
+    // detach user and company if required
+    await handleDetachUserAndCompany(contactId, event);
+  }
+
+  return getResponse(method, endpoint, headers, payload);
+};
+
+const constructTrackResponse = (event) => {
+  const { destination, metadata } = event;
+  const apiVersion = getApiVersion(destination);
+  const payload = transformTrackPayload(event);
+  const method = 'POST';
+  const endpoint = `${getBaseEndpoint(destination)}/events`;
+  const headers = getHeaders(metadata, apiVersion);
+
+  return getResponse(method, endpoint, headers, payload);
+};
+
+const constructGroupResponse = async (event) => {
+  const { destination, metadata } = event;
+  const apiVersion = getApiVersion(destination);
+  const payload = transformGroupPayload(event);
+
+  const method = 'POST';
+  let endpoint = `${getBaseEndpoint(destination)}/companies`;
+  const headers = getHeaders(metadata, apiVersion);
+  let finalPayload = payload;
+
+  // create or update company
+  const companyId = await createOrUpdateCompany(payload, destination, metadata);
+
+  // when contact is found in intercom
+  const contactId = await searchContact(event);
+  if (contactId) {
+    // attach user and company
+    finalPayload = {
+      id: companyId,
+    };
+    endpoint = `${getBaseEndpoint(destination)}/contacts/${contactId}/companies`;
+    await attachContactToCompany(finalPayload, endpoint, destination, metadata);
+  }
+
+  // add tags to company
+  await addOrUpdateTagsToCompany(companyId, event);
+
+  return getResponse(method, endpoint, headers, finalPayload);
+};
+
+const processEvent = async (event) => {
+  const { message } = event;
+  const messageType = getEventType(message);
+  let response;
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      response = await constructIdentifyResponse(event);
+      break;
+    case EventType.TRACK:
+      response = constructTrackResponse(event);
+      break;
+    case EventType.GROUP:
+      response = await constructGroupResponse(event);
+      break;
+    default:
+      throw new InstrumentationError(`message type ${messageType} is not supported.`);
+  }
+  return response;
+};
+
+const process = async (event) => {
+  const { destination } = event;
+  const apiVersion = getApiVersion(destination);
+  let response;
+  if (apiVersion === 'v2') {
+    response = await processEvent(event);
+  } else {
+    throw new InstrumentationError(`apiVersion ${apiVersion} is not supported.`);
+  }
+  return response;
```

Comment:
If this is a new destination we should not give option of Anything apart from `v2` hence at this point this field in UI config doesnt make sense